codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,286 @@
+            "MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. " +
+                    "Commands that may modify the data set are disabled. Please check Redis logs for details " +
+                    "about the error.",
+            "NOAUTH Authentication required.",
+            "NOREPLICAS Not enough good slaves to write.",
+            "NOSCRIPT No matching script. Please use EVAL.",
+            "OOM command not allowed when used memory > 'maxmemory'.",
+            "READONLY You can't write against a read only slave.",
+            "WRONGTYPE Operation against a key holding the wrong kind of value",
+    };
+
+    static {
+        setPreloadStrings(RedisMessageType.SIMPLE_STRING, DEFAULT_PRELOAD_SIMPLE_STRINGS);
+        setPreloadStrings(RedisMessageType.ERROR, DEFAULT_PRELOAD_ERRORS);
+    }
+
+    private RedisMessageFactory() {
+    }
+
+    /**
+     * Set preload strings. Users can override default strings.
+     *
+     * @param type one of {@link RedisMessageType#SIMPLE_STRING} or {@link RedisMessageType#ERROR}.
+     * @param messages an array of String to preload.
+     */
+    public static void setPreloadStrings(RedisMessageType type, String[] messages) {
+        ObjectUtil.checkNotNull(messages, "messages must not be null");
+        if (type == RedisMessageType.SIMPLE_STRING) {
+            PRELOAD_SIMPLE_STRING = convertPreloadsCache(RedisMessageType.SIMPLE_STRING, messages);
+        } else if (type == RedisMessageType.ERROR) {
+            PRELOAD_ERRORS = convertPreloadsCache(RedisMessageType.ERROR, messages);
+        } else {
+            throw new IllegalArgumentException("not support type: " + type);
+        }
+    }
+
+    private static PreloadStringsCache convertPreloadsCache(RedisMessageType type, String[] messages) {
+        Map<String, PreloadString> stringsCache = new HashMap<String, PreloadString>();
+        Map<ByteBuf, PreloadString> byteBufCache = new HashMap<ByteBuf, PreloadString>();
+        for (String message : messages) {
+            ByteBuf byteBuf = Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8));
+            PreloadString preloadString = new PreloadString(type, message, byteBuf);
+            stringsCache.put(message, preloadString);
+            byteBufCache.put(byteBuf, preloadString);
+        }
+        PreloadStringsCache cache = new PreloadStringsCache(stringsCache, byteBufCache);
+        return cache;
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#SIMPLE_STRING}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createSimpleString(String content) {
+        if (content == null) {
+            throw new IllegalArgumentException("Content must not be null");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Content must not be null" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1192)